src/Leela.cpp

```diff
@@ -205,13 +205,45 @@ static void parse_commandline(int argc, char *argv[]) {
         cfg_gtp_mode = true;
     }
 
+#ifdef USE_OPENCL
+    if (vm.count("gpu")) {
+        cfg_gpus = vm["gpu"].as<std::vector<int> >();
+        // if we use OpenCL, we probably need more threads for the max so that we can saturate the GPU.
+        cfg_max_threads *= cfg_gpus.size();
```

Comment:
This needs to respect MAX_CPUS.

The issue is that the OpenBLAS code has a rather weird threading design, and needs the maximum amount of thread to be set at compile time. It's 64 or 128 for common Linux distributions.

I am a bit annoying that forcing it to not thread (which we do in Network) does not avoid this problem, but from experience it does seem to crash if you exceed the built-in limit.

Reply:
Okay.  Pushed a fix that does it.